### PR TITLE
add i18n to some components in scaffolder plugin

### PR DESCRIPTION
**Pull request by angeliski**
## Hey, I just made a Pull Request!

I was playing a little with the new i18n system so we can boost this in Hacktoberfest (Great ideia from @ahhhndre ), but I think could be nice talk a little about this. So I raise this draft and will send some points to discussion:

### 1. Should a Plugin be full moved to I18N before first merge?
I started to move some strings from components and... OMG threre is a lot of words here 😅 
So my first suggestion is to be clear about this: We want all or nothing?

**Suggestion:** I prefer not wait for the full translation, so we can wave small PRs

### 2. Flat keys vs Deep Keys
I started to translate looking using the User Settings as reference.
But this plugin, like many others have so many components I got a little confused about to stay flat.

Juts to get context, [we have two](https://www.i18next.com/translation-function/essentials#accessing-keys) options here:
```js
{
    "component_name": "My Component",
    "component": {
        "name": "My Component"
    }
}

t('component_name') 
t('component.name')
```
I stayed using flat keys, but after walk a little in the translation ref I got some doubts.
Flat keys will provide a simple way to reuse words and phrases, but can get a little messy.
Deep keys could provide a more organized approach, but we can got a big file because of that.

**Suggestion:** Stay flat and things got messy we can refactor and provide a migration path (maybe a tool too)

### 3. Semantic Keys vs Plain text keys
We have a lot of work to change those hard coded words. So I started to use [this plugin](https://github.com/lokalise/i18n-ally) to help me. It's really fast extract the word to a json and after that put in TS. But the fast way to do that is using a approach more "your key is your text", like [here](https://github.com/backstage/backstage/pull/20130/files#diff-f878076e8e66396eb54ab2b3aa53912df1704b7c79399e4a737b62fd8e1c68c6R68). We can use more semantic text like [here](https://github.com/backstage/backstage/pull/20130/files#diff-f878076e8e66396eb54ab2b3aa53912df1704b7c79399e4a737b62fd8e1c68c6R65) too, it's just more slow because we need to get context and understand a little about the component.

**Suggestion:** Stay using plain text keys for most cases and when some is really specific change the approach (like [here](https://github.com/backstage/backstage/pull/20130/files#diff-f878076e8e66396eb54ab2b3aa53912df1704b7c79399e4a737b62fd8e1c68c6R48))

### 4. Provide a No-React way to access the I18N instance
We have some cases where text is outside a component (like [here](https://github.com/backstage/backstage/pull/20130/files#diff-4787c2c98e4f01f6a65f5c0310eb34dce0e3d3b5453ad7e9dd9ecdce89b5739cR27)) so we need to put a way to grab the I18N instance so we can call the translate method.

**Suggestion:** Maybe I am missing how to do that, we can provide a doc or maybe put this in a singleton so we can grab that in a simple import

### 5. Be explicit about the Non Translate out-of-box
I was trying to find new languages in the project when I see we don't have a standard way to do that. I can translate may Backstage instance to Portuguese, but just in my code. Could be nice if we give some pointers about next steps about that so we don't get people frustated or expecting to contribuite for a thing and got another.

**Suggestion:** Create a issue with the next steps (and missing points) about the current i18n state in Backstage

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder/src/translation.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { createTranslationRef } from '@backstage/core-plugin-api/alpha';
+
+export const scaffolderTranslationRef = createTranslationRef({
```
Comment (Rugvip) [inline plugins/scaffolder/src/translation.ts:18]: Let's mark this as @alpha and export it via /alpha too

**Review by Rugvip — CHANGES_REQUESTED** [review]
Awesome, thank you! :tada:

**Comment by Rugvip** [comment]
Ah another bit: you can provide translations for other languages to createTranslationRef. This is all still very early, but you can see it in action in tests: https://github.com/backstage/backstage/blob/d4a8a921699aed81db801bf6ff32716ca7c9e8d4/packages/core-plugin-api/src/translation/TranslationResource.test.ts#L64

**Comment by benjdlambert** [comment]
@angeliski I'd also like to push towards adding i18n to things that's in the `next` folder as a priority as those things will hopefully become the default components moving forward after this next mainline release. :pray:

File: plugins/scaffolder/src/translation.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  id: 'scaffolder',
+  messages: {
+    failed_to_load_installed_actions: 'Failed to load installed actions',
```
[the remark this answers is not in the document]
Comment (angeliski) [reply plugins/scaffolder/src/translation.ts:21]: Nice! I will try some tricks with https://eslint.org/docs/latest/rules/sort-keys

**Pull request closed without merging**